### PR TITLE
Adds different logo for verified publishers

### PR DESCRIPTION
**Pull request by NejcZdovc**
Resolves https://github.com/brave/brave-browser/issues/4043

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests && npm run test-security`) on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection

## Test Plan:
- enable rewards
- don't dismiss grant notification
- go to non-verified site
- make sure that bat logo is the same
- go to verified site
- make sure that you see regular logo with notification count
- dismiss notification
- make sure that you see verified bat logo
- go to non-verified site and make sure that you see regular logo

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: components/brave_rewards/resources/extension/brave_rewards/background/reducers/rewards_panel_reducer.ts [hunk 1]
```diff
@@ -6,19 +6,27 @@ import { types } from '../../constants/rewards_panel_types'
 import * as storage from '../storage'
 import { getTabData } from '../api/tabs_api'
 
-function setBadgeText (state: RewardsExtension.State): void {
+function setBadgeText (state: RewardsExtension.State, verified: boolean = false, tabId: number = -1): void {
   let text = ''
 
-  if (state && state.notifications) {
+  if (verified) {
+    text = '✔️'
```
Comment (NejcZdovc) [inline components/brave_rewards/resources/extension/brave_rewards/background/reducers/rewards_panel_reducer.ts]: @rossmoody can you please see if this check is ok or would you like something else. This is how it looks like:

image

Comment (rossmoody) [reply components/brave_rewards/resources/extension/brave_rewards/background/reducers/rewards_panel_reducer.ts]: 👍

File: components/brave_rewards/resources/extension/brave_rewards/utils.ts [hunk 2]
```diff
@@ -58,3 +58,37 @@ export const getGrant = (grant?: RewardsExtension.GrantInfo) => {
...
+  let data: chrome.browserAction.BadgeTextDetails = {
+    text
+  }
+
+  if (tabId !== -1) {
+    data.tabId = tabId
+    chrome.browserAction.setBadgeBackgroundColor({
+      color: verified ? '#4C54D2' : '#FB542B',
+      tabId
+    })
```
Comment (rossmoody) [inline components/brave_rewards/resources/extension/brave_rewards/utils.ts]: Curious: If this ternary sets the badge color to orange in all instances except verified, do we need the `else` section of this? Or is it a failsafe?

File: components/brave_rewards/resources/extension/brave_rewards/utils.ts [hunk 3]
```diff
@@ -58,3 +58,37 @@ export const getGrant = (grant?: RewardsExtension.GrantInfo) => {
 
   return grant
 }
+
+export const setBadgeText = (state: RewardsExtension.State, verified: boolean = false, tabId: number = -1) => {
```
Comment (petemill) [inline components/brave_rewards/resources/extension/brave_rewards/utils.ts]: nit: this is not a generic 'util', plus it's best to keep things that are unrelated in separate files, in case we do chunking or asynchronous dependency imports (`import()`). Plus, it uses app state as a param. I suggest this is just put in its own module `setBadgeText` or, if you want to be more generic then, `browserAction.ts`.

File: components/brave_rewards/resources/extension/brave_rewards/background.ts [hunk 4]
```diff
@@ -21,17 +19,23 @@ const iconOn = {
...
+  active: true,
+  currentWindow: true
+}, (tabs) => {
+  if (!tabs || !tabs.length) {
+    return
+  }
+
+  let tab = tabs[0]
+  if (tab && tab.id) {
+    rewardsPanelActions.init(tab)
```
Comment (petemill) [inline components/brave_rewards/resources/extension/brave_rewards/background.ts]: Why do we need an `init` stage now just for the active window's active tab? Do we depend on `onPublisherData` which happens whenever a tab is loaded? Either way, is there a reason to send `currentWindow: true` which is unreliable from the background script (which is across multiple windows). Presumably we should send multiple tabs and set the badge text/icon for all the active tabs in all windows? In fact, if we have the data, we could just set badge text for all the open tabs at the same time if we wanted to.

File: components/brave_rewards/resources/extension/brave_rewards/utils.ts (hunk 2 above)
Comment (petemill) [reply components/brave_rewards/resources/extension/brave_rewards/utils.ts]: The else is if we are settings the badge text / icon default, and not overriding for a specific tab (to put the verified badge text)

File: components/brave_rewards/resources/extension/brave_rewards/utils.ts (hunk 3 above)
Comment (NejcZdovc) [reply components/brave_rewards/resources/extension/brave_rewards/utils.ts]: done

File: components/brave_rewards/resources/extension/brave_rewards/background.ts (hunk 4 above)
Comment (NejcZdovc) [reply components/brave_rewards/resources/extension/brave_rewards/background.ts]: true, we should set active tab for all windows. Fixed

File: components/brave_rewards/resources/extension/brave_rewards/background/reducers/rewards_panel_reducer.ts [hunk 5]
```diff
@@ -102,9 +88,15 @@ export const rewardsPanelReducer = (state: RewardsExtension.State | undefined, a
         }
 
         publishers[id] = {
-          tabUrl: tab.url
+          tabUrl: tab.url,
+          tabId: tab.id
         }
+      } else if (publisher && publisher.tabUrl === tab.url && publisher.tabId !== tab.id && validKey) {
```
Comment (ryanml) [inline components/brave_rewards/resources/extension/brave_rewards/background/reducers/rewards_panel_reducer.ts]: nit: can we break this one up a little bit :)

Comment (NejcZdovc) [reply components/brave_rewards/resources/extension/brave_rewards/background/reducers/rewards_panel_reducer.ts]: done, let me know if this is what you had in mind

**Comment by NejcZdovc** [comment]
@ryanml lock file removed
